reactive/flannel.py

```diff
@@ -291,6 +291,11 @@ def reset_states_and_redeploy():
         log(str(e))
 
 
+@hook('pre-series-upgrade')
+def pre_series_upgrade():
+    status_set('blocked', 'Series upgrade in progress')
```

Comment:
Seems like this status could easily get overwritten by other handlers. I recommend setting it in an atexit handler or refactoring the charm to use layer-status.